### PR TITLE
specs, clientNamespace

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
fix https://github.com/Azure/cadl-ranch/issues/758

This scenario mainly test the case when `@clientNamespace` applied to TypeSpec Namespace/Interface/Model/Enum.

So far it does not test corner cases, like client/subclient in different namespace, name conflict for model in different namespace, etc.

# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/azure/client-generator-core/client-namespace/client.tsp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  - SecondClientResult: Azure.ClientGenerator.Core.ClientNamespace.Second
+  - SecondClientEnumType: Azure.ClientGenerator.Core.ClientNamespace.Second.Sub
+  """)
+namespace Client;
+
+@client({
+  name: "ClientNamespaceFirstClient",
+  service: _Specs_.Azure.ClientGenerator.Core.ClientNamespace,
+})
+@clientNamespace("azure.clientgenerator.core.clientnamespace", "java")
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/client-generator-core/client-namespace/client.tsp:25]: I assume for other languages:

csharp: `Azure.ClientGenerator.Core.ClientNamespace`
python: `azure.client_generator.core.client_namespace`

Let me know what the pattern for Go/JS. I will fill in here before PR public for review.

File: packages/cadl-ranch-specs/http/azure/client-generator-core/client-namespace/client.tsp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+interface ClientNamespaceFirstClient {
+  get is First.get;
+}
+@@clientNamespace(FirstModel, "azure.clientgenerator.core.clientnamespace.firstt", "java");
```
Comment (msyyc) [inline packages/cadl-ranch-specs/http/azure/client-generator-core/client-namespace/client.tsp:29]:
```suggestion
@@clientNamespace(FirstModel, "azure.clientgenerator.core.clientnamespace.first", "java");
```

File: packages/cadl-ranch-specs/http/azure/client-generator-core/client-namespace/main.tsp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+namespace Second {
+  #suppress "@azure-tools/cadl-ranch-expect/missing-scenario" "scenario defined in client.tsp"
+  @route("/second")
+  @get
+  op get(): Model.SecondClientResult;
+
+  namespace Model {
+    model SecondClientResult {
```
Comment (msyyc) [inline packages/cadl-ranch-specs/http/azure/client-generator-core/client-namespace/main.tsp:30]: I think it is a little weird to put model into another namespace different with client on purpose. For python SDK, we will always put model into folder `models`, then current cadl-ranch will results in namespace `xxx.model.models` in python.

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/client-namespace/main.tsp:30]: Java is same logic, it would be in `model.models` in azure flavor.

However, there is nothing prohibit unbranded from doing this?

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/client-namespace/main.tsp:30]: PS: in client.tsp, I actually map this to same client namespace of the client.
On the other hand, I actually map the `SecondClientEnumType` to a sub namespace.

```ts
@@clientNamespace(Second.Model, "azure.clientgenerator.core.clientnamespace.second", "java");
@@clientNamespace(Second.Model.SecondClientEnumType, "azure.clientgenerator.core.clientnamespace.second.sub", "java");
```

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/client-namespace/main.tsp:30]: @msyyc 

The possible scenario of model/client in difference space (described by Srikanta) is that there be 2 catalogs of models
1. model common to multiple clients
2. model specific on a single client

If we put clients to different namespaces, there will certainly be for "common model", some of them would have different namespace from the client that uses them.

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/client-namespace/main.tsp:30]: And even without this test scenario, lots of existing tests defined model in a different namespace (e.g. https://github.com/microsoft/typespec/pull/5051#discussion_r1845911525)
